dartagnan/src/main/java/com/dat3m/dartagnan/program/event/core/Event.java

```diff
@@ -181,6 +182,28 @@ public void delete() {
 		}
 	}
 
+	public void insertAfter(Event toBeInserted) {
+		if (this.successor != null) {
+			this.successor.setPredecessor(toBeInserted);
+		}
+		this.setSuccessor(toBeInserted);
+	}
+	public void insertAfter(List<Event> toBeInserted) {
+		if (toBeInserted.isEmpty()) {
+			return;
+		}
+
+		if (this.successor != null) {
+			this.successor.setPredecessor(toBeInserted.get(toBeInserted.size() - 1));
+		}
+		this.setSuccessor(toBeInserted.get(0));
+	}
```

Comment:
Good point. Currently, all users of that function make sure that the events are already linked, but it may be a good idea to let this function make sure everything is linked as expected.